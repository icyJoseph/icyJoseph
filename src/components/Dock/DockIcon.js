import React from "react";
import PropTypes from "prop-types";
import { curry } from "../../functional";

export const DockIcon = ({ tile, clickHandler }) => {
  const { id, Component, fill } = tile;
  const injectHandlerWithContent = curry(clickHandler)(id);
  return (
    <svg
      style={{
        height: "100px",
        width: "100px",
        margin: "5px",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }}
    >
      <Component handler={injectHandlerWithContent} fill={fill} />
    </svg>
  );
};

export default DockIcon;

DockIcon.propTypes = {
  tile: PropTypes.shape({
    Component: PropTypes.func,
    fill: PropTypes.string,
    backgroundFill: PropTypes.string,
    Content: PropTypes.func
  }),
  clickHandler: PropTypes.func
};
