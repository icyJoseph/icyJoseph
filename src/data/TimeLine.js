export default [
  {
    title: "April, 2018",
    meta: "CodeJam",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog
      The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "March, 2018",
    meta: "Hash Code",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "January, 2018",
    meta: "React Certificate",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "January, 2018",
    meta: "Volvo Group",
    fill: "#C92F00",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "November, 2017",
    meta: "Open Hack Winner",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "October, 2016",
    meta: "Consultant",
    fill: "#292929",
    backgroundFill: "#5159BF",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "June, 2016",
    meta: "M.Sc. in Science with Major in Entepreneurship and Business Design",
    fill: "#292929",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "January, 2016",
    meta: "CTO, Co-founder",
    fill: "#F0DB4F",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "March, 2013",
    meta: "Project Engineer",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "December, 2012",
    meta: "B.Sc. in Science with Major in Electronics Engineering",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  }
];
