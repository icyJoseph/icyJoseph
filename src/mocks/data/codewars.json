{
  "username": "icyJoseph",
  "name": "Joseph",
  "honor": 663,
  "clan": "Evolve Technology Sweden",
  "leaderboardPosition": 38381,
  "skills": null,
  "ranks": {
    "overall": { "rank": -4, "name": "4 kyu", "color": "blue", "score": 1158 },
    "languages": {
      "rust": { "rank": -4, "name": "4 kyu", "color": "blue", "score": 1158 }
    }
  },
  "codeChallenges": { "totalAuthored": 0, "totalCompleted": 28 }
}
