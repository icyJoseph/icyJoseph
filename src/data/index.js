import React from "react";
import Loadable from "react-loadable";
import Spinner from "../components/Loading/Spinner";
import Loading from "../logos/Loading";
import TimeLine from "./TimeLine";

export default [
  {
    id: 0,
    title: "About Me",
    meta: "Joseph",
    Component: Loadable({
      loader: () => import("../logos/Me"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/About"),
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 1,
    title: "ReactJS",
    meta: "Certified",
    Component: Loadable({
      loader: () => import("../logos/ReactLogo"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#292929",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="react" type="repos" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 2,
    title: "JavaScript",
    meta: "Favorite Language",
    background: "#193549",
    Component: Loadable({
      loader: () => import("../logos/JavaScriptLogo"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#F0DB4F",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Recast"),
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 3,
    title: "Redux",
    meta: "Framework",
    Component: Loadable({
      loader: () => import("../logos/Redux"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#292929",
    background: "#193549",
    backgroundFill: "#5159BF",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="redux" type="repos" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 4,
    title: "algorithms",
    meta: "Problem Solving",
    Component: Loadable({
      loader: () => import("../logos/Problem"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="algorithms" type="solutions" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 5,
    title: "experience",
    meta: "My career",
    Component: Loadable({
      loader: () => import("../logos/Experience"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/TimeLine"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} experience data={TimeLine} />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 6,
    title: "python",
    meta: "Code as a hobby",
    Component: Loadable({
      loader: () => import("../logos/PythonLogo"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="python" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 7,
    title: "hackathon",
    meta: "Value Creation",
    Component: Loadable({
      loader: () => import("../logos/Rocket"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="hackathon" type="hacks" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 8,
    title: "awards",
    meta: "Acknowledgments",
    Component: Loadable({
      loader: () => import("../logos/Award"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#C92F00",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Common"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} title={"awards"} />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 9,
    title: "currently",
    meta: "Life as a React Developer",
    Component: Loadable({
      loader: () => import("../logos/Report"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Common"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} title={"currently"} />;
      },
      loading() {
        return <Spinner />;
      }
    })
  }
];
