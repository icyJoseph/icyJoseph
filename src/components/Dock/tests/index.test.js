import React from "react";
import { shallow } from "enzyme";
import { Dock } from "../";
import { DockIcon } from "../DockIcon";
import data from "../../../data";

describe("DockIcon", () => {
  const spy = jest.fn();
  const props = {
    tile: {
      ...data[0]
    },
    clickHandler: spy
  };

  const wrapper = shallow(<DockIcon {...props} />);

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("injects children with clickHandler", () => {
    const onClick = wrapper
      .children()
      .at(0)
      .prop(["handler"]);
    expect(onClick).toBeInstanceOf(Function);
    onClick();
    expect(spy).toHaveBeenCalledTimes(1);
  });
});

describe("Dock", () => {
  const items = data.filter(d => !d.hide);

  const props = {
    items,
    clickHandler: jest.fn()
  };

  const wrapper = shallow(<Dock {...props} />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders all items", () => {
    expect(wrapper.find(DockIcon)).toHaveLength(items.length);
  });
});
