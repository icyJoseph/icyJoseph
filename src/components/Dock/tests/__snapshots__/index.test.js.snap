// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dock renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dock
    clickHandler={[MockFunction]}
    items={
      Array [
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 0,
          "meta": "Joseph",
          "title": "About Me",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#292929",
          "id": 1,
          "meta": "Certified",
          "title": "ReactJS",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#F0DB4F",
          "id": 2,
          "meta": "Favorite Language",
          "title": "JavaScript",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "#5159BF",
          "fill": "#292929",
          "id": 3,
          "meta": "Framework",
          "title": "Redux",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 4,
          "meta": "Problem Solving",
          "title": "algorithms",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 6,
          "meta": "Code as a hobby",
          "title": "python",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 7,
          "meta": "Value Creation",
          "title": "hackathon",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DockIcon
          clickHandler={[MockFunction]}
          tile={
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 0,
              "meta": "Joseph",
              "title": "About Me",
            }
          }
        />,
        <DockIcon
          clickHandler={[MockFunction]}
          tile={
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#292929",
              "id": 1,
              "meta": "Certified",
              "title": "ReactJS",
            }
          }
        />,
        <DockIcon
          clickHandler={[MockFunction]}
          tile={
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#F0DB4F",
              "id": 2,
              "meta": "Favorite Language",
              "title": "JavaScript",
            }
          }
        />,
        <DockIcon
          clickHandler={[MockFunction]}
          tile={
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "#5159BF",
              "fill": "#292929",
              "id": 3,
              "meta": "Framework",
              "title": "Redux",
            }
          }
        />,
        <DockIcon
          clickHandler={[MockFunction]}
          tile={
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 4,
              "meta": "Problem Solving",
              "title": "algorithms",
            }
          }
        />,
        <DockIcon
          clickHandler={[MockFunction]}
          tile={
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 6,
              "meta": "Code as a hobby",
              "title": "python",
            }
          }
        />,
        <DockIcon
          clickHandler={[MockFunction]}
          tile={
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 7,
              "meta": "Value Creation",
              "title": "hackathon",
            }
          }
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "tile": Object {
            "Component": [Function],
            "Content": [Function],
            "background": "#193549",
            "backgroundFill": "tomato",
            "fill": "#273B7A",
            "id": 0,
            "meta": "Joseph",
            "title": "About Me",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "tile": Object {
            "Component": [Function],
            "Content": [Function],
            "background": "#193549",
            "backgroundFill": "tomato",
            "fill": "#292929",
            "id": 1,
            "meta": "Certified",
            "title": "ReactJS",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "tile": Object {
            "Component": [Function],
            "Content": [Function],
            "background": "#193549",
            "backgroundFill": "tomato",
            "fill": "#F0DB4F",
            "id": 2,
            "meta": "Favorite Language",
            "title": "JavaScript",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "tile": Object {
            "Component": [Function],
            "Content": [Function],
            "background": "#193549",
            "backgroundFill": "#5159BF",
            "fill": "#292929",
            "id": 3,
            "meta": "Framework",
            "title": "Redux",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "tile": Object {
            "Component": [Function],
            "Content": [Function],
            "background": "#193549",
            "backgroundFill": "tomato",
            "fill": "#273B7A",
            "id": 4,
            "meta": "Problem Solving",
            "title": "algorithms",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "tile": Object {
            "Component": [Function],
            "Content": [Function],
            "background": "#193549",
            "backgroundFill": "tomato",
            "fill": "#273B7A",
            "id": 6,
            "meta": "Code as a hobby",
            "title": "python",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "tile": Object {
            "Component": [Function],
            "Content": [Function],
            "background": "#193549",
            "backgroundFill": "tomato",
            "fill": "#273B7A",
            "id": 7,
            "meta": "Value Creation",
            "title": "hackathon",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DockIcon
            clickHandler={[MockFunction]}
            tile={
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 0,
                "meta": "Joseph",
                "title": "About Me",
              }
            }
          />,
          <DockIcon
            clickHandler={[MockFunction]}
            tile={
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#292929",
                "id": 1,
                "meta": "Certified",
                "title": "ReactJS",
              }
            }
          />,
          <DockIcon
            clickHandler={[MockFunction]}
            tile={
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#F0DB4F",
                "id": 2,
                "meta": "Favorite Language",
                "title": "JavaScript",
              }
            }
          />,
          <DockIcon
            clickHandler={[MockFunction]}
            tile={
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "#5159BF",
                "fill": "#292929",
                "id": 3,
                "meta": "Framework",
                "title": "Redux",
              }
            }
          />,
          <DockIcon
            clickHandler={[MockFunction]}
            tile={
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 4,
                "meta": "Problem Solving",
                "title": "algorithms",
              }
            }
          />,
          <DockIcon
            clickHandler={[MockFunction]}
            tile={
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 6,
                "meta": "Code as a hobby",
                "title": "python",
              }
            }
          />,
          <DockIcon
            clickHandler={[MockFunction]}
            tile={
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 7,
                "meta": "Value Creation",
                "title": "hackathon",
              }
            }
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "display": "flex",
          "flexWrap": "wrap",
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "tile": Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 0,
              "meta": "Joseph",
              "title": "About Me",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "tile": Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#292929",
              "id": 1,
              "meta": "Certified",
              "title": "ReactJS",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "tile": Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#F0DB4F",
              "id": 2,
              "meta": "Favorite Language",
              "title": "JavaScript",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "tile": Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "#5159BF",
              "fill": "#292929",
              "id": 3,
              "meta": "Framework",
              "title": "Redux",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "tile": Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 4,
              "meta": "Problem Solving",
              "title": "algorithms",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "tile": Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 6,
              "meta": "Code as a hobby",
              "title": "python",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "tile": Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 7,
              "meta": "Value Creation",
              "title": "hackathon",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`DockIcon renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DockIcon
    clickHandler={[MockFunction]}
    tile={
      Object {
        "Component": [Function],
        "Content": [Function],
        "background": "#193549",
        "backgroundFill": "tomato",
        "fill": "#273B7A",
        "id": 0,
        "meta": "Joseph",
        "title": "About Me",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <LoadableComponent
        fill="#273B7A"
        handler={[Function]}
      />,
      "style": Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100px",
        "justifyContent": "center",
        "margin": "5px",
        "width": "100px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "fill": "#273B7A",
        "handler": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <LoadableComponent
          fill="#273B7A"
          handler={[Function]}
        />,
        "style": Object {
          "alignItems": "center",
          "display": "flex",
          "height": "100px",
          "justifyContent": "center",
          "margin": "5px",
          "width": "100px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fill": "#273B7A",
          "handler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
